src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -21,10 +26,25 @@ const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
       }
     })
 
+    // TODO: appStateChange triggers on the web platform as well, using visibilitychange listener, consider reusing the code(downside @capacitor/app dependency on web & extension)
+    const appStateChangeListenerHandle = App.addListener('appStateChange', ({ isActive }) => {
```

Comment:
I would expect `lockAsync` to handle that. If there is an unlocked state, lock it. Otherwise ignore. And I tested it, and it works as expected. I could technically skip the `lockAsync` call, but I don't see a reason to.

Reply:
What is a point in initialize locking code for not opened accounts?

Reply:
You could technically create profile from empty state, which would require you to add this listener dynamically. Or if the profile is removed you would need to remove the listener. 

Isn't it easier to just attach to the `appStateChange` event once? We could ignore the `lockAsync` call, I agree on that. But again IMO `lockAsync` should be able to handle it(like it does, with current implementation).

Reply:
yeah I think you are right and with this simplification mobile code does not spread across app